### ISSUE TITLE
version 1.0.0

### ISSUE THREAD

**Issue by theScrabi** [TeamNewPipe/NewPipe#857]
I think after all we accomplished so far we can think about releasing version 1.0.0
There are still several things missing:
- Support for livestream
- Suptitle support
- bookmarks
- homestreaming support

But all in all the impotent functions NewPipe has to offer are implemented, and everything feels sort of like a decent app that supports YouTube. So I think we can now consider to put a 1 at the beginning of our version number :)

Just to remember, my plan or roadmap for NewPipe was to accomplish a good working YouTube support until version 1.0 and a good working multi service support until version 2.0.
So by branding NewPipe with 1.0 we can then focuse (more) on multi service implementation.

what do you guys think about that?

**Comment by wb9688** [comment]
>Support for livestream

That won't be hard to add I think. At least, if ExoMedia decently supports it. I know ExoPlayer can though. I could add support to NewPipeExtractor for it, but I'm already busy with other things, so…

>bookmarks

I'd prefer a local playlists system over a bookmarks system.

>Just to remember, my plan or roadmap for NewPipe was to accomplish a good working YouTube support until version 1.0 and a good working multi service support until version 2.0.

Yeah, _was_. IMHO we should first finish adding support for multiple services in NewPipe, eg. I (or maybe somebody else) should finish #693. I however agree at only adding support for other services to NewPipeExtractor should be done after v1.0.0, well unless somebody else contributes support for another service.

I think we should make a GitHub project ([here](https://github.com/TeamNewPipe/NewPipe/projects)) to track what we want in v1.0.0.

**Comment by TobiGr** [comment]
IMHO before releasing 1.0 we should

- create tours so new users know about hidden features and new features can be easily introduced
- use a navigation drawer instead of the 3 dot menu in the app bar

to provide a good user experience. I think releasing 1.0 is going to call attention to many people who blog about NewPipe and bring new users [and donors :)] to our app. I think both thing can be done quite quickly without too much effort.

**Comment by wb9688** [comment]
@TobiGr: Yeah, but your point 2 would be part of that #693 I mentioned.

**Comment by dcoder-mm** [comment]
I just wanna say that NewPipe is the best alternative youtube app i've ever seen. Great work, really. 

Think it missing some little details. For example - auto continue playing in background when switching or hiding app (like official youtube client). And some details about interface (eg. unsubscribing directly from subscribtion list, w\o going to channel page)

--
Also now i'm trying to implement playlist\queue downloading feature. Just for practice my coding skills, mostly. What do you think about this feature?

**Comment by theScrabi** [comment]
> - use a navigation drawer instead of the 3 dot menu in the app bar

I would do this after 1.0 since I want to put the multi service list into this drawer. By releasing 1.0 with an almos empty drawer would make the app not feel whole anymore.

> auto continue playing in background when switching or hiding app 

Well 1.0 does not mean that no improvements will come after ;)

**Comment by avently** [comment]
@dcoder-mm
> auto continue playing in background when switching or hiding app
It's already done

**Comment by TobiGr** [comment]
> I would do this after 1.0 since I want to put the multi service list into this drawer. By releasing 1.0 with an almos empty drawer would make the app not feel whole anymore.

@theScrabi Yes. You're right

> Yeah, but your point 2 would be part of that #693 I mentioned.

@wb9688 Perfect. I answered via phone so I didn't see your answer.

> I think we should make a GitHub project (here) to track what we want in v1.0.0.

That's a good idea. _Off-topic:_ I think we should also create milestones/projects for any "bigger" release and keep the almost weekly release rhythm we had once to quickly deploy bugfixes.

**Comment by wb9688** [comment]
>I answered via phone so I didn't see your answer.

@TobiGr: No problem.

>I think we should also create milestones/projects for any "bigger" release and keep the almost weekly release rhythm we had once to quickly deploy bugfixes.

@TobiGr: I agree, but we should really get the beta stuff working.

@theScrabi: What's the current state of the beta stuff? I.e. is it set up at the F-Droid side?

**Comment by theScrabi** [comment]
> What's the current state of the beta stuff? I.e. is it set up at the F-Droid side?

Not yet. I wanted to work on it tho.

**Comment by TobiGr** [comment]
And what about the press kit (#667)? I think it we should at least release some basic information, so blogs and magazines can get verified information and screenshots/media about NewPipe.

**Comment by tgp1994** [comment]
I think this sounds great. It's been fun being a part of NewPipe's development and watching it grow. I agree that YouTube functionally should be implemented by 1.0.

**Comment by r2308145** [comment]
> There are still several things missing:
> -     Support for livestream
> -     Suptitle support
> -     bookmarks
> -     homestreaming support

**Comment by Cormak** [comment]
Most of the features listed in the OP seem like 2.0 features.

I think we should just focus on the smaller things right now, like:

- Keep a video playing even when the screen is turned off. 

- Emulate the original Youtube app by playing video in the original video space in portrait mode while allowing users to scroll through other videos instead of just going to fullscreen.

- Add a means of notifying the user when a new video shows up in their subscriptions when the app is opened.

- Enable sound on all video downloads. 

These are all useful things that new users already expect from Newpipe, but don't have yet.

**Comment by TobiGr** [comment]
@theScrabi What do you think of releasing a minor version with all bug fixes included now and targeting version 1.0.0 for the end of this year / with c34?

**Comment by wb9688** [comment]
I agree there should be a minor version with the bugs fixed ASAP.

**Comment by r2308145** [comment]
indeed, I ride this boat with you guys.. there are currently small bugs which ruins the NewPipe experience a bit

**Comment by theScrabi** [comment]
Yea for 1.0 i didn't think to add more features, but making it stable.

**Comment by Cormak** [comment]
@theScrabi Not only should you make it stable, you shouldn't focus on the big features until all the little features I listed above are added.

**Comment by theScrabi** [comment]
@Cormak 
As I said, NewPipe feels like a descent YouTube app, and that's enough for a 1.0.0. Just saying there is a version 1 there does not mean there will be no features afterwards.

Focusing on shifting over to multi service shouldn't be stretched out for to long, since that is its getting more and more complicated the more features we add.
Adding smaller features will always be important, but they can be implemented.

Also @Cormak some of your features have already been widely discussed, and if you think they are important please open up a separate issue for each of them.

**Comment by theScrabi** [comment]
@wb9688 @TobiGr @tgp1994 I agree with you. I'll release in the next few days.

**Comment by Cormak** [comment]
@theScrabi I know that there's going to be more stuff added after 1.0.0 but I really think you should take care of the smaller stuff I mentioned first before moving onto bigger things. As you said, it's getting more complicated, the more features you add, so it makes the most sense to do the little features that people have been wanting for a long time first before moving onto something big.

**Comment by theScrabi** [comment]
I released [v0.11.1](https://github.com/TeamNewPipe/NewPipe/releases/tag/v0.11.1) so we can now focus on debuging until version 1.0

**Comment by TobiGr** [comment]
@theScrabi We should open a project to collect all issues across all Repositories and then assign devs to the issues to see who is working on what.

**Comment by theScrabi** [comment]
All right sounds good.

**Comment by avently** [comment]
Any news?

**Comment by TobiGr** [comment]
Take a look at our [project page](https://github.com/TeamNewPipe/NewPipe/projects).

**Comment by theScrabi** [comment]
0.1.5 felt almost bugfree (at least to me) how about we keep the features low now, and make the next version version 1.0? So all other features can than be part of the beta version.

**Comment by avently** [comment]
@theScrabi I agree. NewPipe should be 1.0 now. There are so many feature in it

**Comment by speednroll** [comment]
It would be great to launch 1.0 but unfortunately I do not agree. I still have a couple of mistakes. For example, when I use the "what's new" section I receive an error for sent by mail and some other small errors in the channels

**Comment by speednroll** [comment]
I'm not sure if the background player was unexpectedly closed again like in previous versions (I'm not sure why that happened to me) and there's still the bug that when you press a video on the results and play it, it does not appear in the history, nor the videos of the playlists. if these errors are solved, it would be happy :D

**Comment by tgp1994** [comment]
I have to agree with @speednroll , there are still [a number of open bugs](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aopen+is%3Aissue+label%3Abug) for NewPipe. I might be biased because one of those bugs personally affects me (where NewPipe has started disregarding the t parameter). In at least one instance, NewPipe is less functional (working) than it was before. :(

While I respect that it is ultimately your decision @theScrabi , I'd like to strongly encourage some more bug bashing before a 1.0 release.

**Comment by wb9688** [comment]
I'm OK with a 1.0 release, when first local playlist support is implemented
(as v0.12) and then actively hunt for bugs in NewPipe, fix them, release
v0.12.1, wait a week (after it's in F-Droid), look if there's any problem
reported on Sentry or GitHub and when we're sure it has no real bugs
anymore, release v1.0.

**Comment by speednroll** [comment]
I totally agree with the opinion of @wb9688

**Comment by theScrabi** [comment]
@wb9688 I was willing to release 1.0 asap.
The Idea was that once 1.0 is release new features will be include into the beta app which will be released simultaneously with 1.0. Because the problem is that with every new feature we release we also introcude new bugs. So I wanted to keep the features added before 1.0 low, and than put new features into beta before they are well tested.

This however makes it necesary to release 1.0 ASAP.

So I would not put local playlists into 1.0 but do it afterwards,
OR release beta before 1.0 gets to be introduced (would be another option).

**Comment by wb9688** [comment]
@theScrabi: Ah, OK, then I'd say the same, but without "first local
playlist support is implemented (as v0.12) and then" and with v0.11.6
instead of v0.12.1.

**Comment by theScrabi** [comment]
Actually I am looking for new bugs on sentry right now. If  v0.15.0 does does not trigger any (non android segmentation related) errors id relabel it as 1.0.

Does that sound good?

**Comment by tgp1994** [comment]
What about the bugs that haven't been fixed yet?

**Comment by theScrabi** [comment]
Yea I just see there are still many.

Maybe we should consider to release beta before 1.0.

**Comment by TobiGr** [comment]
@theScrabi I'd suggest to fix #861 #1009 #940 before version 1.0 and release a beta with 1.0. They shouldn't be that hard to fix. Many of these bugs are outdated e.g. #815 and #737 can be closed.

**Comment by theScrabi** [comment]
Well my fear is that we will continue to encounter bugs. This will not end, or at least take a long time.

**Comment by tgp1994** [comment]
@theScrabi You can assign bugs on GitHub to certain milestones. So what you could do, for example, is tag the above bugs with a "beta" milestone, meaning they'll be fixed before a beta is released. That way you can set a defined goal for what needs to be done before another release.

**Comment by avently** [comment]
@theScrabi i can fix #1009 TobiGr wrote in my PR. Its pretty easy to do because I can save history of videos in one place - in one service. In your current source code it will be a little tricky. So you can leave this bug after 1.0 or we can merge my PR and local playlists before 1.0. Yes, second choice maybe will introduce new bugs too.

**Comment by theScrabi** [comment]
Well for this reason we introduced [this](https://github.com/TeamNewPipe/NewPipe/projects/3) at our project page.

**Comment by theScrabi** [comment]
So actually I want to release beta ASAP, does not matter when 1.0 is released. The reason is because we also have a lot of feature PR which can not be merged because of this 1.0 problem.

I'll set something up.

**Comment by theScrabi** [comment]
So [here](https://gitlab.com/derSchabi/fdroiddata/commit/2236d9c4565cc85d3419f5a09f64f4bfa3e95d9e) is my current setup for building NewPipe beta version.

**Comment by avently** [comment]
@theScrabi did you finish your exams?

**Comment by theScrabi** [comment]
Not yet.

**Comment by TobiGr** [comment]
> So here is my current setup for building NewPipe beta version.

@theScrabi Looks good to me, but I didn't tested it.

**Comment by theScrabi** [comment]
I did. Works.

**Comment by theScrabi** [comment]
> @theScrabi You can assign bugs on GitHub to certain milestones. So what you could do, for example, is tag the above bugs with a "beta" milestone, meaning they'll be fixed before a beta is released. That way you can set a defined goal for what needs to be done before another release.

We don't have milestones, but we do have an [kanban board for this](https://github.com/TeamNewPipe/NewPipe/projects/3).

**Comment by tgp1994** [comment]
@theScrabi Perfect. So any subsequent bugs that get submitted won't interfere with your release unless one of them is a critical type of bug.

**Comment by theScrabi** [comment]
You could say it like this yes.

**Comment by theScrabi** [comment]
So as I said I changed the strategy, and release the beta version before version 1.0 is ready.

**Comment by ale5000-git** [comment]
@theScrabi: Could you please put beta versions on F-Droid?

**Comment by theScrabi** [comment]
Its comming to fdroid. it just takes a little while.

**Comment by ale5000-git** [comment]
Hi, what is the current status?

**Comment by theScrabi** [comment]
Far away from it. NewPipe is bug more than ever.

**Comment by opusforlife2** [comment]
Newpipe has changed drastically since then. A new issue should be opened whenever this is being considered again.
